wandb/sdk/launch/utils.py

```diff
@@ -154,7 +154,8 @@ def construct_launch_spec(
 
     if entry_point:
         launch_spec["overrides"]["entry_point"] = entry_point
-    launch_spec["cuda"] = cuda
+    if cuda is not None:
```

Comment:
this needs to be reverted, we have this complicated cuda handling where a None indicates unspecified and so must be set. (this is to handle the case where cuda is preserved from the previous run but if the user wants to set a non-cuda run, they set cuda=False. if cuda=None then cuda is preserved)

Reply:
maybe good to add a comment here saying see the handling in cli.launch

Reply:
Thinking through the cases here:
1. Previous run cuda, no cuda flag set anywhere -> default to previous run. Since launch_project.cuda will be None this will work.
2. Previous run cuda, cuda false flag set in cli call -> sets it to false, won't use cuda
3. Previous run not cuda, cuda flag true -> sets cuda to true
4. otherwise use cuda flag set in launch spec.
I think this is correct as is, I'll check/add tests

Reply:
I've added tests to verify this behavior

Reply:
ok yeah this works because we have a `launch_spec.get("cuda", None)` in create_project_from_spec. nice